maps/tether/submaps/underdark_pois/underdark_things.dm

```diff
@@ -0,0 +1,251 @@
+							/obj/item/weapon/ore/diamond,
+							/obj/item/weapon/ore/diamond,
+							/obj/item/weapon/ore/diamond,
+							/obj/item/weapon/ore/diamond,
+							/obj/item/weapon/ore/diamond,
+							/obj/item/weapon/ore/diamond,
+							/obj/item/weapon/ore/diamond,
+							/obj/item/weapon/ore/diamond
+							),
+				prob(1);list(
+							/obj/item/weapon/mining_scanner,
+							/obj/item/weapon/shovel/spade,
+							/obj/item/weapon/ore/verdantium,
+							/obj/item/weapon/ore/verdantium,
+							/obj/item/weapon/ore/verdantium,
+							/obj/item/weapon/ore/verdantium,
+							/obj/item/weapon/ore/verdantium
+							)
+				)
+
+/obj/random/multiple/underdark/treasure
+	name = "random underdark treasure"
+	desc = "Random treasure loot for Underdark."
+	icon = 'icons/obj/storage.dmi'
+	icon_state = "cashbag"
+
+/obj/random/multiple/underdark/treasure/item_to_spawn()
+	return pick(
+				prob(5);list(
+							/obj/random/coin,
+							/obj/random/coin,
+							/obj/random/coin,
+							/obj/random/coin,
+							/obj/random/coin,
+							/obj/item/clothing/head/pirate
+							),
+				prob(4);list(
+							/obj/item/weapon/storage/bag/cash,
+							/obj/item/weapon/spacecash/c500,
+							/obj/item/weapon/spacecash/c100,
+							/obj/item/weapon/spacecash/c50
+							),
+				prob(3);list(
+							/obj/item/clothing/head/hardhat/orange,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold,
+							/obj/item/stack/material/gold),
+				prob(1);list(
+							/obj/item/stack/material/phoron,
+							/obj/item/stack/material/phoron,
+							/obj/item/stack/material/phoron,
+							/obj/item/stack/material/phoron,
+							/obj/item/stack/material/diamond,
+							/obj/item/stack/material/diamond,
+							/obj/item/stack/material/diamond
+							)
+				)
+
+/obj/random/multiple/underdark/mechtool
+	name = "random underdark mech equipment"
+	desc = "Random mech equipment loot for Underdark."
+	icon = 'icons/mecha/mecha_equipment.dmi'
+	icon_state = "mecha_clamp"
+
+/obj/random/multiple/underdark/mechtool/item_to_spawn()
+	return pick(
+				prob(12);list(/obj/item/mecha_parts/mecha_equipment/tool/drill),
+				prob(10);list(/obj/item/mecha_parts/mecha_equipment/tool/hydraulic_clamp),
+				prob(8);list(/obj/item/mecha_parts/mecha_equipment/generator),
+				prob(7);list(/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/scattershot/rigged),
+				prob(6);list(/obj/item/mecha_parts/mecha_equipment/repair_droid),
+				prob(4);list(/obj/item/mecha_parts/mecha_equipment/combat_shield),
+				prob(2);list(/obj/item/mecha_parts/mecha_equipment/weapon/energy/riggedlaser),
+				prob(1);list(/obj/item/mecha_parts/mecha_equipment/wormhole_generator),
```

Comment:
I would replace the wormhole generator and combat shield with the other rigged weapons (such as the LMG).